src/nni_manager/training_service/kubernetes/kubernetesTrainingService.ts

```diff
@@ -329,37 +332,31 @@ abstract class KubernetesTrainingService {
         );
         return registrySecretName;
     }
-
-    protected async uploadFilesToAzureStorage(trialJobId: string, trialLocalTempFolder: string, codeDir: string, uploadRetryCount: number | undefined): Promise<string> {
+    
+    /**
+     * upload local directory to azureStorage
+     * @param srcDirectory the source directory of local folder
+     * @param destDirectory the target directory in azure
+     * @param uploadRetryCount the retry time when upload failed
+     */
+    protected async uploadFolderToAzureStorage(srcDirectory: string, destDirectory: string, uploadRetryCount: number | undefined): Promise<string> {
         if (this.azureStorageClient === undefined) {
             throw new Error('azureStorageClient is not initialized');
         }
-        let trialJobOutputUrl: string = '';
         let retryCount: number = 1;
         if(uploadRetryCount) {
             retryCount = uploadRetryCount;
         }
-        let resultUploadNNIScript: boolean = false;
-        let resultUploadCodeFile: boolean = false;
+        let uploadSuccess: boolean = false;
+        const folderUriInAzure = `https://${this.azureStorageAccountName}.file.core.windows.net/${this.azureStorageShare}/${destDirectory}`;
         try {
             do {
-                //upload local files, including scripts for running the trial and configuration (e.g., hyperparameters) for the trial, to azure storage
-                if(!resultUploadNNIScript) {
-                    resultUploadNNIScript = await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
-                        `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
-                        `${trialLocalTempFolder}`);
-                }
-                //upload code files to azure storage
-                if(!resultUploadCodeFile) {
-                    resultUploadCodeFile = await AzureStorageClientUtility.uploadDirectory(this.azureStorageClient,
-                        `nni/${getExperimentId()}/${trialJobId}`, this.azureStorageShare,
-                        `${codeDir}`);
-                }
-                if (resultUploadNNIScript && resultUploadCodeFile) {
-                    trialJobOutputUrl = `https://${this.azureStorageAccountName}.file.core.windows.net/${this.azureStorageShare}` + 
-                    `/${path.join('nni', getExperimentId(), trialJobId, 'output')}`;
-                    break;
-                } else {
+                uploadSuccess = await AzureStorageClientUtility.uploadDirectory(
+                    this.azureStorageClient,
+                    `${destDirectory}`, 
+                    this.azureStorageShare,
+                    `${srcDirectory}`);
+                if (!uploadSuccess) {
                     //wait for 5 seconds to re-upload files
                     await delay(5000);
```

Comment:
there is no `break` in the modified version

Reply:
fixed.